### ISSUE TITLE
Sword Specialization is resetting the swing timer

### ISSUE THREAD

**Issue by magey** [magey/tbc-warrior#31]
Sword Specialization was [changed in patch 2.1.3](https://gist.github.com/magey/6beb165ea93cdb010d5f22a21a9db5c2#file-patch-txt-L3916) to not reset your swing timer when it procs:

```
  * Sword Specialization: The change to Sword Specialization making its
    extra attacks appear in yellow has been reverted. Extra attacks will
    appear in white and act like any auto-attack. They will no longer
    reset the swing time of your weapon. Rage awarded will be calculated
    by damage dealt as with any normal auto-attack. 
```

Currently on the beta it still resets your swing timer. Example log (abridged version):

```
3/27 04:47:05.307  SWING_DAMAGE
3/27 04:47:05.307  SWING_DAMAGE_LANDED
3/27 04:47:06.104  SPELL_CAST_SUCCESS,"Hamstring"
3/27 04:47:06.104  SPELL_EXTRA_ATTACKS,"Sword Specialization"
3/27 04:47:06.104  SPELL_DAMAGE,"Hamstring"
3/27 04:47:06.104  SPELL_AURA_REFRESH,"Hamstring"
3/27 04:47:06.104  SWING_DAMAGE
3/27 04:47:06.104  SWING_DAMAGE_LANDED
3/27 04:47:06.104  SPELL_AURA_REFRESH,"Deep Wound"
3/27 04:47:08.222  SWING_DAMAGE
3/27 04:47:08.222  SWING_DAMAGE_LANDED
```

In this log I'm using a 2.1 speed Gladius. Normal swing happens at `04:47:05.307`, followed by a Sword Specialization proc off of Hamstring at `04:47:06.104`. The next swing should occur at `04:47:05.307 + 2.1 = 04:47:07.407` but instead it happens at `04:47:08.222` which is 2.116s after the Sword Specialization extra attack.

**Comment by Scyllaqt** [comment]
still occuring in build 2.5.1.38339

**Comment by magey** [comment]
Fixed today in 2.5.1.38521:

![image](https://user-images.githubusercontent.com/44882404/116633582-9795d400-a962-11eb-9275-acfbbe7101bb.png)

Using a 3.70 sword, we can see a normal swing at 34.621 and then a Hamstring cast at 37.569 which proc'd Sword Specialization. The next normal swing occurs at 38.325 which is 3.704s after the previous normal swing, meaning the swing timer was unaffected by the extra attack.
